src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length < 1) {
+      return '';
+    }
+
+    let result = '';
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        result += '';
+        result += separator;
+      } else {
+        result += this[i];
+        result += separator;
+      }
```

Comment:
В каждом условии пишешь result += separator, его можно вынести

```suggestion
      if (this[i] === null || this[i] === undefined) {
        result += '';
      } else {
        result += this[i];
      }
      result += separator;
```

Reply:
Согласен